core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -256,9 +266,13 @@ final class ShuffleBlockFetcherIterator(
     // Add the remote requests into our queue in a random order
     fetchRequests ++= Utils.randomize(remoteRequests)
 
+    assert ((0 == reqsInFlight) == (0 == bytesInFlight),
+      "reqsInFlight = " + reqsInFlight + ", bytesInFlight = " + bytesInFlight)
     // Send out initial requests for blocks, up to our maxBytesInFlight
     while (fetchRequests.nonEmpty &&
-      (bytesInFlight == 0 || bytesInFlight + fetchRequests.front.size <= maxBytesInFlight)) {
+      (bytesInFlight == 0 || 
```

Comment:
it would be nice to make another function to do this set of checks since its the same as below